server/internal/controllers/users.go

```diff
@@ -273,28 +273,34 @@ func UserPurge(c *gin.Context) {
 		return
 	}
 
-	ok, _, _, _ := auth.AuthenticateUserOfChain(c, db, " ", query.UserUID)
+	ok, user, _, _ := auth.AuthenticateUserOfChain(c, db, "", query.UserUID)
 	if !ok {
 		return
 	}
 
-	var userID uint
-	db.Raw("SELECT id FROM users WHERE uid = ? LIMIT 1", query.UserUID).Scan(&userID)
-
-	var email string
-	db.Raw("SELECT email FROM users WHERE uid = ? LIMIT 1", query.UserUID).Scan(&email)
-
-	var testChain uint
-	db.Raw(
-		`SELECT chain_id FROM user_chains WHERE user_id = ? 
-		AND is_chain_admin = 1 
-		HAVING COUNT(is_chain_admin) = 1;`,
-		userID).Scan(&testChain)
-	db.Exec(`DELETE FROM user_chains WHERE chain_id = ?`, testChain)
-	db.Exec(`DElETE FROM chains WHERE id = ?`, testChain)
+	// find chains where user is the last chain admin
+	chainIDsToDelete := []uint{}
```

Comment:
Note how I'm using `[]uint{}` this defines and initializes `chainIDsToDelete`

GoLang is tricky as basic types default to `0` or and empty string while pointers and slices default to `nil`

```go
var i int // => 0
var s string // => ""
var p *string // => nil
var a []int // => nil
var o struct{ID int} // => {ID: 0}

i := 0 // => 0
s := "" // => ""
p := &s // => *("")
a := []int{1,2} // => [1,2]
o := struct{ID int}{ID: 1} // => {ID: 1}
```